cpp/src/replace/replace.cu

```diff
@@ -13,31 +16,528 @@
+    if (threadIdx.x < cudf::experimental::detail::warp_size && 0 == (threadIdx.x % cudf::experimental::detail::warp_size)) {
+      atomicAdd(output_valid_count, block_valid_count);
+    }
+  }
+}
+
+  /* --------------------------------------------------------------------------*/
+  /**
+   * @brief Functor called by the `type_dispatcher` in order to invoke and instantiate
+   *        `replace_kernel` with the appropriate data types.
+   */
+  /* ----------------------------------------------------------------------------*/
+  struct replace_kernel_forwarder {
+    template <typename col_type,
+              std::enable_if_t<cudf::is_fixed_width<col_type>()>* = nullptr>
+    void operator()(cudf::column_view const& input_col,
+                    cudf::column_view const& values_to_replace,
+                    cudf::column_view const& replacement_values,
+                    cudf::mutable_column_view& output,
+                    cudaStream_t stream = 0)
+    {
+      cudf::size_type *valid_count = nullptr;
+      rmm::device_scalar<cudf::size_type> valid_counter(0);
+      if (output.nullable()) {
+        valid_count = valid_counter.data();
+      }
+
+      cudf::experimental::detail::grid_1d grid{output.size(), BLOCK_SIZE, 1};
+
+      auto replace = replace_kernel<col_type, true, true>;
+
+      if (input_col.has_nulls()){
+        if (replacement_values.has_nulls()){
+          replace = replace_kernel<col_type, true, true>;
+        }else{
+          replace = replace_kernel<col_type, true, false>;
+        }
+      }else{
+        if (replacement_values.has_nulls()){
+          replace = replace_kernel<col_type, false, true>;
+        }else{
+          replace = replace_kernel<col_type, false, false>;
+        }
+      }
+
+      auto device_in = cudf::column_device_view::create(input_col);
+      auto device_out = cudf::mutable_column_device_view::create(output);
+      auto device_values_to_replace = cudf::column_device_view::create(values_to_replace);
+      auto device_replacement_values = cudf::column_device_view::create(replacement_values);
+
+      replace<<<grid.num_blocks, BLOCK_SIZE, 0, stream>>>(
+                                             *device_in,
+                                             *device_out,
+                                             valid_count,
+                                             output.size(),
+                                             *device_values_to_replace,
+                                             *device_replacement_values);
+
+      cudaStreamSynchronize(stream);
+
+      if(valid_count != nullptr){
+        cudf::size_type valids {0};
+        CUDA_TRY(cudaMemcpyAsync(&valids,
+                                 valid_count,
+                                 sizeof(cudf::size_type),
+                                 cudaMemcpyDefault,
+                                 stream));
+        output.set_null_count(output.size() - valids);
```

Comment:
This only changes the null count of the view, not the column itself. This is another reason this function should return the column instead of accept a `mutable_view` argument.

Reply:
Changed so that it is actually setting the null count of the column itself.